### PR TITLE
Arc 1556 safe guard on subscription find x

### PR DESCRIPTION
**Pull request by gxueatlassian**
**What's in this PR?**
Adding safe guard to the `Subscriptions` model, ( and accidentally add support for `gitHubAppId` as well)

**Why**
If `GHE` accidentally turned on, the safe guard of `gitHubAppId is null` to protect data from messing up between `cloud` and `server`.

**Added feature flags**
N/A

**Affected issues**  
https://softwareteams.atlassian.net/browse/ARC-1556

**How has this been tested?**  
Unit test.
Stage: Tested

**Whats Next?**
Refer to linked issue under ticket https://softwareteams.atlassian.net/browse/ARC-1480

### CODE REVIEW

File: src/models/subscription.ts [hunk 1]
```diff
@@ -79,21 +79,37 @@ export class Subscription extends Model {
...
 		return this.findAll({
 			where: {
-				gitHubInstallationId: installationId
+				gitHubInstallationId: installationId,
+				...(gitHubAppId ? {
+					gitHubAppId
+				} : {
+					gitHubAppId: {
+						[Op.is]: null
+					}
```
Comment (mboudreau) [inline src/models/subscription.ts:93]: can't this just be represented with 
```
this.findAll({
			where: {
				gitHubInstallationId: installationId,
        gitHubAppId: gitHubAppId || null
	}
});
```
?

File: src/models/subscription.ts [hunk 2]
```diff
@@ -79,21 +79,37 @@ export class Subscription extends Model {
...
 		return this.findOne({
 			where: {
-				gitHubInstallationId: gitHubInstallationId
+				gitHubInstallationId: gitHubInstallationId,
+				...(gitHubAppId ? {
+					gitHubAppId
+				} : {
+					gitHubAppId: {
+						[Op.is]: null
+					}
```
Comment (mboudreau) [inline src/models/subscription.ts:111]: same here

File: src/models/subscription.ts [hunk 3]
```diff
@@ -135,6 +152,16 @@ export class Subscription extends Model {
...
 		}
 
+		andFilter.push({
+			...(gitHubAppId ? {
+				gitHubAppId
+			} : {
+				gitHubAppId: {
+					[Op.is]: null
+				}
+			})
```
Comment (mboudreau) [inline src/models/subscription.ts:162]: same here

File: src/models/subscription.ts [hunk 4]
```diff
@@ -155,24 +182,20 @@ export class Subscription extends Model {
...
-		return this.findOne({
-			where: {
-				jiraClientKey: clientKey,
-				gitHubInstallationId: installationId
+				gitHubInstallationId,
+				...(gitHubAppId ? {
+					gitHubAppId
+				} : {
+					gitHubAppId: {
+						[Op.is]: null
```
Comment (mboudreau) [inline src/models/subscription.ts:196]: same here

File: src/models/subscription.ts [hunk 5]
```diff
@@ -196,7 +219,14 @@ export class Subscription extends Model {
...
 			where: {
 				gitHubInstallationId: payload.installationId,
-				jiraHost: payload.host
+				jiraHost: payload.host,
+				...(payload.gitHubAppId ? {
+					gitHubAppId: payload.gitHubAppId
+				} : {
+					gitHubAppId: {
+						[Op.is]: null
+					}
```
Comment (mboudreau) [inline src/models/subscription.ts:228]: same here

File: src/models/subscription.ts (hunk 1 above)
Comment (gxueatlassian) [reply src/models/subscription.ts]: I have no idea what I was doing, lol.
fixed.

File: src/models/subscription.ts (hunk 3 above)
Comment (gxueatlassian) [reply src/models/subscription.ts]: fixed.

File: src/models/subscription.ts (hunk 4 above)
Comment (gxueatlassian) [reply src/models/subscription.ts]: fixed

File: src/models/subscription.ts (hunk 5 above)
Comment (gxueatlassian) [reply src/models/subscription.ts]: fixed

File: src/models/subscription.ts (hunk 2 above)
Comment (gxueatlassian) [reply src/models/subscription.ts]: fixed

**Review by mboudreau — APPROVED** [review]

**Pull request merged**